tests/component-plugin/src/main/java/org/mule/functional/api/component/StacktraceLogChecker.java

```diff
@@ -172,34 +208,7 @@ public String toString() {
 
     @Override
     public int hashCode() {
-      int methodHashCode = method != null ? method.hashCode() : MISSING_DATA_HASH_CODE;
-      int packageHashCode = packageName != null ? packageName.hashCode() : MISSING_DATA_HASH_CODE;
-      int classHashCode = clazz != null ? clazz.hashCode() : MISSING_DATA_HASH_CODE;
-      int lineHashCode = lineNumber != null ? lineNumber.hashCode() : MISSING_DATA_HASH_CODE;
-      return methodHashCode + packageHashCode + classHashCode + lineHashCode;
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-      if (!(obj instanceof MethodCall)) {
-        return false;
-      }
-      if (this == obj) {
-        return true;
-      }
-      if (method != null && ((MethodCall) obj).method != null && !method.equals(((MethodCall) obj).method)) {
-        return false;
-      }
-      if (clazz != null && ((MethodCall) obj).clazz != null && !clazz.equals(((MethodCall) obj).clazz)) {
-        return false;
-      }
-      if (packageName != null && ((MethodCall) obj).packageName != null && !packageName.equals(((MethodCall) obj).packageName)) {
-        return false;
-      }
-      if (lineNumber != null && ((MethodCall) obj).lineNumber != null && !lineNumber.equals(((MethodCall) obj).lineNumber)) {
-        return false;
-      }
-      return true;
+      return Objects.hashCode(method) + Objects.hashCode(packageName) + Objects.hashCode(clazz) + Objects.hashCode(lineNumber);
```

Comment:
hashCode receives all the objects to use in the calculation. You don't need to add